woocommerce/woocommerce-android
# Free Trial: Make store profiler flow switchable

Summary
==========
This PR adds a `Store Profiler` Feature Flag that allows us to switch when using the Store Profiler group of views working alongside the Free Trial feature. This aims to enable us to easily switch between activating the Store Profiler or not without worrying about changing actual behaviors as things get experimented with.

Also, this PR centralizes the Free Trial store creation logic inside a single Use Case class called `CreateFreeTrialStore` that can be easily reused in any Store Creation ViewModel with proper unit test coverage.

Captures
==========

### Store Profiler **ENABLED**

| Free Trial Flow | Standard Store Creation Flow |
| ------------- | ------------- |
| https://user-images.githubusercontent.com/5920403/229667527-48192106-d9f1-49b0-9599-04f1d37c2467.mp4 | https://user-images.githubusercontent.com/5920403/229667653-470b3db4-8147-48e6-8f22-6383fde2a669.mp4 |

### Store Profiler **DISABLED**

| Free Trial Flow | Standard Store Creation Flow |
| ------------- | ------------- |
| https://user-images.githubusercontent.com/5920403/229667451-f94d3256-ad8b-449b-923a-ceacf26ecd46.mp4 | https://user-images.githubusercontent.com/5920403/229667588-173b660f-bead-424f-97a9-fb610b1504ac.mp4 |


How to Test
==========
Adjust the `FeatureFlag` class to activate and deactivate the Free Trial and Store Profiler features to see if they follow the expected configuration.

### Scenario 1 - Free Trial **ENABLED** and Store Profiler **ENABLED**
### Scenario 2 - Free Trial **ENABLED** and Store Profiler **DISABLED**
### Scenario 3 - Free Trial **DISABLED** and Store Profiler **ENABLED**
### Scenario 4 - Free Trial **DISABLED** and Store Profiler **DISABLED**


Update release notes:
